File: 02. JavaScript/01. DSA/01. Language Wise Practice/02. Python/01. Bit Manipulation/01. FirstSetBit.py
def get_first_set_bit(n):
  if n == 0:
    return 0
  position = 1
  while n > 0:
    if (n&1) == 1:
      return position
    
    n = n >> 1
    position += 1



inputs = [18, 20, 3365, 15, 0]
for num in inputs:
  print(f"First set bit of {num} is at position: {get_first_set_bit(num)}")

File: 01. Data Structures and Algorithms/01. Language Wise Practice/01. JavaScript/01. Bit Manipulation/01. FirstSetBit.js
const getFirstSetBit = (n) => {
  if(n === 0){
    return 0;
  }
  let position = 1;
  while(n>0){
    if((n&1) === 1){
      return position;
    }
    n = n >> 1;
    position++;
  }
};

const inputs = [18, 20, 3365, 15, 0];

inputs.forEach((num) => {
  console.log(`First set bit of ${num} is at position: ${getFirstSetBit(num)}`)
})